lhotse/workflows/activity_detection/base.py

```diff
@@ -0,0 +1,132 @@
+import abc
+import multiprocessing
+from concurrent.futures import ProcessPoolExecutor
+from dataclasses import dataclass
+from functools import partial
+from itertools import chain
+from typing import Dict, List, Optional, Type
+
+import numpy as np
+
+from lhotse.audio.recording import Recording
+from lhotse.audio.recording_set import RecordingSet
+from lhotse.supervision import SupervisionSegment, SupervisionSet
+
+
+@dataclass
+class Activity:
+    start: float
+    duration: float
+
+
+def _format_sampling_rate(sampling_rate: int) -> str:
+    formatted_rate = str(round(sampling_rate / 1000, 1))
+    formatted_rate = formatted_rate.replace(".0", "")
+    return f"{formatted_rate}kHz"
+
+
+class ActivityDetector(abc.ABC):
+    def __init__(self, detector_name: str, sampling_rate: int, device: str = "cpu"):
+        self._name = f"{detector_name}_{_format_sampling_rate(sampling_rate)}"
+        self._sampling_rate = sampling_rate
+        self._device = device
+
+    @property
+    def device(self) -> str:
+        return self._device
+
+    def __call__(self, recording: Recording) -> List[SupervisionSegment]:
+        resampled = recording.resample(self._sampling_rate)
+        audio = resampled.load_audio()  # type: ignore
+
+        uid_template = "{recording_id}-{detector_name}-{channel}-{number:05}"
```

Comment:
Perhaps we can remove the detector name from the segment ID to keep it concise? Users can always just save the manifest file with the appropriate name.

Reply:
I suggest to leave the `detector_name` in the segment identifier to allow mapping of segments created by `SileroVAD` to segments of other detectors that may be implemented in the future. 

Suppose you want to use a detector that detects the sound of a cell phone ringing. Currently, you can easily create your own detector by inheriting it from the `lhotse.workflows.acivity_detection:ActivityDetector` class and overriding the `forward` method to make it fit your needs. You can then take advantage of all the functionality associated with the activity detection task. For example, you can use your detector to segment the dataset, converting the results into a `SupervisionSegment`. This functionality is already implemented in the `__call__` method of the ActivityDetector class. 

Thus, storing the detector name in the segment identifier provides flexibility for further expansion and integration of other detectors.

Reply:
Are any changes required on this issue?